### ISSUE TITLE
Login to /auth/token/login fails because the path became more strict in patch release 2.2.1

### ISSUE THREAD

**Issue by dhrp** [sunscrapers/djoser#778]
Release 2.2.1 introduced (but not documented) a change to the url routes. 

Specifically, the following test used to pass, but now fails.
```
        response = client.post(
            "/auth/token/login",
            {"email": "testuser@acme.com", "password": "some-password"},
            format="json",
        )
        self.assertEqual(response.status_code, 200)
```

Luckily I had a test to warn me, but this change, introduced as a patch release has the potential to break actual logins in the wild.

Let me also be clear: Using this path for login is not "django standard" -> they have a trailing slash; but front-end developers may not be aware and may have used the path without a trailing slash.

The actual change is here:
https://github.com/sunscrapers/djoser/compare/2.2.0...2.2.1#diff-46b93deeea84b407c499c260c1d4dad6a9ae0868d3da10080d1c09d9ad10f67bL6

Depending on how much you think this is an issue for users I suggest reverting the change in 2.2.2 and introducing it as a 2.3.0. (I auto-upgrade dependencies on path releases)

**Comment by tomwojcik** [comment]
Right, that's my regression, sorry. 

https://github.com/sunscrapers/djoser/commit/16f406e980056dc8768b01c85c912e8e26e4324b#diff-46b93deeea84b407c499c260c1d4dad6a9ae0868d3da10080d1c09d9ad10f67bL6

I did not notice it's `?`, `$` and `?$`. It should have been a single pattern from the beginning :/ Since I'm not sure how to clean it up, I will just roll back these changes.

**Comment by tomwojcik** [comment]
Released in 2.2.2
